cpp/tests/types/dispatcher_test.cu

```diff
@@ -13,122 +13,156 @@
  * See the License for the specific language governing permissions and
  * limitations under the License.
  */
+
+// These tests excerise the `assert(false)` on unsupported dtypes in the
+// type_dispatcher The assert is only present if the NDEBUG macro isn't defined
+#ifdef NDEBUG
+#undef NDEBUG
+#define REDEFINE
+#endif
```

Comment:
This is fishy. Why are you trying to mess with the semantics of `assert()` ? You can always do things like `exit(EXIT_FAILURE)` if you really must - or just return an error.

Reply:
I'm trying to ensure that when you pass an unsupported `gdf_dtype` into the `type_dispatcher` that it hits the error case and exits the program via the `assert` in the default case. https://github.com/rapidsai/cudf/blob/branch-0.6/cpp/src/utilities/type_dispatcher.hpp#L110

If it's a Release build, then `NDEBUG` is defined, and my test will fail because the `assert` won't be called and therefore the program won't exit.

I agree it's not an ideal solution, but I couldn't come up with anything better to exercise the error cases of `type_dispatcher`.

Reply:
But why are you trying to do that via `assert()`? Programmers _rely_ on the semantics of this macro - and of the `NDEBUG` macro as well. You shouldn't change their values. Just look at the man page (on most Linux distros) for `assert` (emphasis mine):
```
If the macro NDEBUG is defined at the moment <assert.h> was last included,
the macro assert() generates no code, and hence does nothing at all. It is 
**not recommended** to define NDEBUG if using assert() to detect error
conditions since the software may behave non-deterministically.
```

I think it would be safer and clearer if you just spelled out what you want to have happen for Release builds. Have a look at the definition of `assert()` if you want to replicate its behavior.